code/controllers/subsystem/fire_burning.dm

```diff
@@ -27,9 +27,10 @@ SUBSYSTEM_DEF(fire_burning)
 				return
 			continue
 
-		if(O.resistance_flags & ON_FIRE)
+		if((O.resistance_flags & ON_FIRE) && !(O.resistance_flags & FIRE_PROOF))
 			O.take_damage(20, BURN, "fire", 0)
 		else
+			O.resistance_flags &= ~ON_FIRE //in case we fireproof while burning
```

Comment:
this is already handled at object level.

Reply:
Or are you saying I should call extinguish() here

Reply:
Nothing should be setting that flag directly as it's supposed to always be handled by extinguished(). If anything is actually doing it, that's a bug that needs to be fixed.

Reply:
Gotcha

Reply:
I'll have to clear this up tomorrow, though, it's late uk time.